src/lemmas.k.md

```diff
@@ -177,50 +168,82 @@ to match on a few different expressions for each write operation.
 The ordering of arguments differs between the various cases to match the bytecode produced by
 solidity.
 
-##### Write `reserve0`
-
-Mask target range to zero:
+##### Zeroing
 
 ```k
 // all regions potentially non-zero
-rule (notMaxUInt112 &Int ((Z *Int pow224) +Int ((Y *Int pow112) +Int X))) => ((Z *Int pow224) +Int (Y *Int pow112))
-  requires #rangeUInt(112, X)
-  andBool #rangeUInt(112, Y)
-  andBool #rangeUInt(32, Z)
+rule X &Int ((C *Int pow224) +Int ((B *Int pow112) +Int A)) => X &Int #WordPackUInt112UInt112UInt32(A, B, C)
```

Comment:
are the following rules needed? are they all covering the constructor case?

Reply:
Maybe they aren't, you're right. After the `SLOAD` we have `#WordPack` on the stack, and then it's `AND`ed with a `maxUInt`.